core/src/main/java/org/opensearch/sql/data/model/ExprDateValue.java

```diff
@@ -25,20 +32,44 @@
 public class ExprDateValue extends AbstractExprValue {
 
   private final LocalDate date;
+  private String datePattern;
+  // List of cleaned patterns
+  private static final List<String> CLEANED_PATTERNS = Arrays.stream(DATE_TIME_FORMATTER_PATTERNS_NANOS_OPTIONAL.split("\\]\\["))
+          .map(pattern -> pattern.replace("[", "").replace("]", ""))
+          .collect(Collectors.toList());
 
   /** Constructor of ExprDateValue. */
   public ExprDateValue(String date) {
     try {
       this.date = LocalDate.parse(date, DATE_TIME_FORMATTER_VARIABLE_NANOS_OPTIONAL);
+      this.datePattern = determineDatePattern(date);
     } catch (DateTimeParseException e) {
       throw new SemanticCheckException(
           String.format("date:%s in unsupported format, please use 'yyyy-MM-dd'", date));
     }
   }
 
+  private String determineDatePattern(String date) {
+    for (String pattern : CLEANED_PATTERNS) {
+      try {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(pattern, Locale.ROOT)
+                .withResolverStyle(ResolverStyle.STRICT);
+        LocalDate.parse(date, formatter);
+        return pattern;
+      } catch (DateTimeParseException e) {
+        // Ignore and try next pattern
+      }
+    }
+    return DateTimeFormatter.ISO_LOCAL_DATE.toString();
+  }
+
   @Override
   public String value() {
-    return DateTimeFormatter.ISO_LOCAL_DATE.format(date);
+    if (this.datePattern == null) {
+      return DateTimeFormatter.ISO_LOCAL_DATE.format(date);
+    }
+    DateTimeFormatter formatter = DateTimeFormatter.ofPattern(this.datePattern , Locale.ROOT);
+    return this.date.format(formatter);
```

Comment:
what is the issue solved by this? could u add a test to cover this change?

Reply:
When we read the value for a literal while building query to opensearch DSL, we are reading the `ExprDateValue.value()` function which by default always converts the local date object into ISO string format. This leads to `OpenSearchParseException ` as it fails to parse date field in some custom format (dd-MM-yy) with ISO format. 
Here I'm formatting the parsed date to the matching input string pattern while converting it to literals in the request.

Reply:
Thx, 
* I understand the reason of add pattern to support parse literal string value.
* But, why we want to value() function keep original format?

Reply:
As discussed offline, when translate Date value to string in term query, we could explore use field format defined in index.

Reply:
What in my mind is why we don't know the customized date format during catalog binding? For example, in analyzing the field as a date, the format pattern "dd-MMM-yy" should be recognized and stored in SQL's `Date` dataType.
```
PUT testindex
{
  "mappings" : {
    "properties" :  {
      "release_date" : {
        "type" : "date",
        "format" : "dd-MMM-yy"
      }
    }
  }
}
```

Reply:
Yes I've updated the PR to use OpenSearch date formatters of a field from its mappings info to parse the date string from Lucene query.